File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int n = nums.length;
        int m = nums[0];
        int sum = nums[0];
        
        for (int i = 1; i < n; i++) {
            if (nums[i] > nums[i - 1]) {
                m += nums[i];
            } else {
                m = nums[i];
            }
            sum = Math.max(sum, m);
        }
        
        return sum;
    }
}

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int n=nums.size();
        int m=nums[0];
        int sum=nums[0];
        for(int i=1;i<n;i++){
            if(nums[i]>nums[i-1])
            m+=nums[i];
            else 
            m=nums[i];
             sum=max(sum,m);
        }
        return sum;
    }
};